pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -555,6 +554,17 @@ func (nim *nodeInfoManager) installDriverToCSINode(
 		TopologyKeys: topologyKeys.List(),
 	}
 
+	if attachVolumeLimitEnabled {
+		if maxAttachLimit > 0 && maxAttachLimit <= math.MaxInt32 {
+			m := int32(maxAttachLimit)
+			driverSpec.Allocatable = &storagev1beta1.VolumeLimits{
+				Count: &m,
+			}
+		} else {
+			klog.Errorf("Invalid attach limit value %d cannot be added to CSINode object for %q", maxAttachLimit, driverName)
```

Comment:
It's technically valid from the CSI spec.  Maybe either truncate the limit to the MaxInt32 or keep it as no limit and change the log to a warning with "Exceeded max supported attach limit value..."